cpp/include/rapidsmpf/streaming/core/base_chunk.hpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cstdint>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace rapidsmpf::streaming {
+
+/**
+ * @brief Common base for streaming chunks.
+ *
+ * Holds the per-chunk sequence number (for ordering) and the CUDA stream
+ * on which the chunk's resources were created.
+ *
+ */
+class BaseChunk {
+  public:
+    /**
+     * @brief Construct a base chunk.
+     *
+     * @param sequence_number Ordering identifier for the chunk.
+     * @param stream CUDA stream associated with the chunk's resources.
+     */
+    constexpr BaseChunk(
+        std::uint64_t sequence_number, rmm::cuda_stream_view stream
+    ) noexcept
+        : sequence_number_{sequence_number}, stream_{stream} {}
+
+    /**
+     * @brief Virtual destructor.
+     */
+    virtual ~BaseChunk() = default;
```

Comment:
*question*: Does this class need to be virtual, or could it "just" be a mixin instead?

It seems like we never need to be able to store polymorphic child classes of `BaseChunk` in the same container...